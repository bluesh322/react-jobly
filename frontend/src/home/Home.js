import React, { useContext } from "react";
import { Link } from "react-router-dom";
import UserContext from "../auth/UserContext";
import { Container, Box, Button } from "@material-ui/core";

const Home = () => {
  const { currentUser } = useContext(UserContext);
  return (
    <Container maxWidth="md">
      <h2 data-testid="resolved">Jobly</h2>
      <p>All the jobs in one, convenient place</p>
      {currentUser ? (
        <h2 data-testid="loggedInResolved" >Welcome Back {currentUser.username}</h2>
      ) : (
          <Box>
          <Link to="/signup" key="Sign Up">
          <Button variant="contained" color="primary">Sign Up</Button>
          </Link>
          <Link to="/login" key="Login">
            <Button variant="contained" color="secondary">Login</Button>
          </Link>              
          </Box>
      )}
    </Container>
  );
};

export default Home;
