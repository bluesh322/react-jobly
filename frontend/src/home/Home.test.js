import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import Home from "./Home";
import TestUserProvider from "../auth/testUser";


it("matches snapshot", function () {
  const { asFragment } = render(
      <MemoryRouter>
        <TestUserProvider>
          <Home />
        </TestUserProvider>
      </MemoryRouter>,
  );
  expect(asFragment()).toMatchSnapshot();
});

it("matches snapshot when logged out", function () {
  const { asFragment } = render(
      <MemoryRouter>
        <TestUserProvider currentUser={null}>
          <Home />
        </TestUserProvider>
      </MemoryRouter>,
  );
  expect(asFragment()).toMatchSnapshot();
});