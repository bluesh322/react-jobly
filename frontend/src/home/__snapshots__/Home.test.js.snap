// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root MuiContainer-maxWidthMd"
  >
    <h2
      data-testid="resolved"
    >
      Jobly
    </h2>
    <p>
      All the jobs in one, convenient place
    </p>
    <h2
      data-testid="loggedInResolved"
    >
      Welcome Back testuser
    </h2>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot when logged out 1`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root MuiContainer-maxWidthMd"
  >
    <h2
      data-testid="resolved"
    >
      Jobly
    </h2>
    <p>
      All the jobs in one, convenient place
    </p>
    <div
      class="MuiBox-root MuiBox-root-1"
    >
      <a
        href="/signup"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Sign Up
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </a>
      <a
        href="/login"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Login
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </a>
    </div>
  </div>
</DocumentFragment>
`;
